README.md

```diff
@@ -136,18 +142,6 @@ val releaseVersion     : SettingKey[String => String]
 val releaseNextVersion : SettingKey[String => String]
 ```
 
-The default settings make use of the helper class [`Version`](https://github.com/sbt/sbt-release/blob/master/src/main/scala/Version.scala) that ships with *sbt-release*.
-
-```scala
-// strip the qualifier off the input version, eg. 1.2.1-SNAPSHOT -> 1.2.1
-releaseVersion     := { ver => Version(ver).map(_.withoutQualifier.string).getOrElse(versionFormatError(ver)) }
-
-// bump the version and append '-SNAPSHOT', eg. 1.2.1 -> 1.3.0-SNAPSHOT
-releaseNextVersion := {
-  ver => Version(ver).map(_.bump(releaseVersionBump.value).asSnapshot.string).getOrElse(versionFormatError(ver))
-},
-```
-
```

Comment:
As mentioned on Discord, my guess is that @xuwei-k's primary concern is breaking this contract that's been around for 10+ years in existing builds in the wild. In other words, you should expect that there are builds out there that literally have those customization e.g.:

```scala
// strip the qualifier off the input version, eg. 1.2.1-SNAPSHOT -> 1.2.1
releaseVersion     := { ver => Version(ver).map(_.withoutQualifier.string).getOrElse(versionFormatError(ver)) }
```

and the reasonable expectation is that this plugin should continue to function _with_ those customization, until it decides to break the compatibility. So if there's breaking change in the README, it's a red flag.

Reply:
Thanks for the quick review!

I don't see this as breaking contract - simply making it more honest and user-friendly. This PR does not remove or change the type signatures of any public entities. Nor does it prevent users from overriding these changes with their own values (i.e. the code linked in the README above will still work and produce exactly the same result as before, without a prerelease increment). The only change is that prerelease versions will, by default, be incremented.

From a user perspective, I see `sbtrelease` saying that it increments versions by default to the `Next` version. I assumed that this meant that the last version number is incremented, including any prerelease version number. But `sbtrelease` does not increment prerelease version numbers; it simply removes the prerelease altogether behind the scenes. Call it part of the established contract over 10 years, but (to me, at least, and I suspect to other users as well) it's unintuitive. Not a bug, but the contract is (slightly) lying to you. 

Obviously if you're creating a numbered prerelease, your default expectation is that there will be multiple of them, not that the prerelease goes to final after the first one. That's where this PR comes in - a `version.sbt` with 1.0.0-RC2-SNAPSHOT by default sets `releaseVersion` to `1.0.0-RC2`, instead of `1.0.0`, and `releaseNextVersion` to `1.0.0-RC3-SNAPSHOT`, instead of `1.0.1-SNAPSHOT`. That's it.

**Note**: I marked `.withoutQualifier` as deprecated (but I didn't remove it), because I noticed that in this code base as well as in others it is just being used to remove the snapshot. Hence the new `.withoutSnapshot` method and the note encouraging users to migrate. If that's controversial, then the two methods can simply live side by side.

Reply:
Here's what I'd suggest (and and also if similar situation comes up in the future): please open a GitHub issue with **steps** (`1.0.0-RC2-SNAPSHOT`), **problems** (it doesn't bump to RC3), and **expectations** (it bumps to RC3). Hopefully the issue would clarify the problem domain, which you rightly feel should be fixed and the existing user probably has not cared since sbt-release has not bumped it all this time. My guess is that current users have been either using SNAPSHOT or full patch versions, without RC cycle.

Although I have no horse in this race, I am generally +1 on supporting this M1/RC1 use case for sbt-release, so long as we can document the breaking/migration aspect clearly in README and release note. Instead of removing the old customization from the README (or deprecating), I suggest we document it like:

> ### RC cycle support
> sbt-release x.y.z adds RC cycle support. Customize the keys as follows: 
> ....
> ### users who don't need RC cycle
> you can keep using the old settings: 
> ....

This way 100% of the existing users who are not using RC cycle can keep their old settings?

Reply:
I've created an issue here, https://github.com/sbt/sbt-release/issues/398, but I think there's some confusion about my changes to the README. Before my PR, this 8-line section in the README [here](https://github.com/sbt/sbt-release/blob/master/README.md?plain=1#L141) is just describing the default behavior by literally copying and pasting the current code base [here](https://github.com/sbt/sbt-release/blob/master/src/main/scala/ReleasePlugin.scala#L226). I've made changes to the README because it shouldn't reproduce the code base. This defeats the purpose of a README - if I want to read the code, of course I don't need a README to do that, but it's also not a maintainable pattern in general, nor is it the most readable. 

So what I have removed from the README did not represent a customization (which would still work in the same way if users actually copied and pasted that in), but just the current default behavior.

That said, are we fine with prerelease increments becoming part of the new default behavior? Ironically, if yes and we want to add a note to help users keep the old settings, I will need to re-include the README code snippet, since it won't reproduce the code base anymore.